src/stores/games.ts

```diff
@@ -1,73 +1,72 @@
 import { computed, ref, type Ref } from 'vue'
 import { defineStore } from 'pinia'
 import { parse_rec } from 'aoe2rec-js'
-import { mapNames } from '@/entities/maps'
-import { civNames } from '@/entities/civs'
+import { Game, Replay, type ParsedReplay } from '@/entities/game'
 
-type Replay = {
-  zheader: {
-    game_settings: {
-      resolved_map_id: number
-      rms_strings: string[]
-      players: {
-        name: string
-        profile_id: string
-        civ_id: number
-      }[]
+export const useGamesStore = defineStore('games', () => {
+  const recs: Ref<Record<string, ParsedReplay>> = ref({})
+  const games: Ref<Game[]> = ref([])
+  const gameCount: Ref<number> = ref(3)
+
+  async function addRec(file: File) {
+    const rec = await parseRec(file)
+    games.value = games.value.filter((game) => !game.isDummy())
+
+    const game = new Game([new Replay(file, rec)])
+    games.value = [...games.value, game].sort(
+      (game1, game2) => (game1?.date?.getTime() ?? 0) - (game2?.date?.getTime() ?? 0)
+    )
+    if (games.value.length > gameCount.value) {
+      setGamesNumber(games.value.length)
+    } else {
+      setGamesNumber(gameCount.value)
     }
-    timestamp: number
   }
-}
 
-export const useGamesStore = defineStore('games', () => {
-  const games: Ref<Record<string, Replay>> = ref({})
-  async function parseGame(file: File) {
-    const reader = new FileReader()
-    reader.addEventListener(
-      'loadend',
-      (event) => {
-        if (!event.target || !event.target.result) {
-          console.log('Could not find target')
-          return
-        }
-        try {
-          const rec: Replay = parse_rec(event.target.result as ArrayBuffer)
-          games.value = { ...games.value, [file.name]: rec }
-        } catch (error) {
-          console.error('Failed to parse')
-          console.error(error)
-        }
-      },
-      false
-    )
-    reader.readAsArrayBuffer(file)
+  async function parseRec(file: File) {
+    return new Promise<ParsedReplay>((resolve) => {
+      const reader = new FileReader()
+      reader.addEventListener(
+        'loadend',
+        (event) => {
+          if (!event.target || !event.target.result) {
+            console.log('Could not find target')
```

Comment:
Should this not be an error?

Reply:
Technically it can't happen, AFAIK.... It's more to make typescript happy

Reply:
Yeah I agree. Does not look like it can happen but let's either change it to `.error()` or remove it altogether.